nvflare/apis/wf_controller_spec.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Callable, List, Union
+
+
+class WFControllerSpec(ABC):
+    @abstractmethod
+    def run(self):
+        """Main `run` routine for the controller workflow."""
+        raise NotImplementedError
+
+    def send_model(
+        self,
+        task_name: str,
+        data: any,
```

Comment:
Should this be `any` or `FLModel`?

Reply:
We can have this base method call "send" so it is not tied to `FLModel`
and use `any`.

And then in the ModelController, we implement both `send` and `send_model` method?

So other classes remain the same change as @SYangster  already have?

Reply:
Right `any` is used since ModelController implements the spec with `FLModel`, but another implementation could use something different. I agree with changing the base method name of `send_model` (however `send` is already a method of Controller which could be confusing)

Reply:
got it, if there's conflict this current name is good.